tools/lkl/include/lkl.h

```diff
@@ -72,12 +72,18 @@ void lkl_perror(char *msg, int err);
 /**
  * lkl_disk - host disk handle
  *
+ * @dev - a pointer to 'virtio_blk_dev' structure for this disk
  * @fd - a POSIX file descriptor that can be used by preadv/pwritev
  * @handle - an NT file handle that can be used by ReadFile/WriteFile
  */
 union lkl_disk {
-	int fd;
-	void *handle;
+	struct {
+		void *dev;
+		union {
+			int fd;
+			void *handle;
+		};
+	};
 };
```

Comment:
No need to keep the union, lets convert lkl_disk to struct. Also, I think you can squish the two commits into a single one.